### PR TITLE
feat: cold db overhaul

### PR DESCRIPTION
**Pull request by wacban**
- Removed unnecessary optimizations from ColdDB. Without those optimizations we can simply implement all of the Database trait method on ColdDB. With fully implemented ColdDB and compatible formats between hot and cold we can support sensible implementation of iterator merging in SplitDB, and, again, implement all of Database trait methods. With SplitDB fully implementing Database there is no risk that some missed unsupported case will break the node. The removed optimizations are: 
  - removing shard uid
  - changing endiannes
  - stripping reference count
- Allow writing to the SplitDB - this is unfortunate but necessary now because the view client (despite "view" in the name) actually writes to the store. So far I've only seen it writing in order to update some cache but it's possible that there are more cases so I opted to just allow it for all tables. 
- Implemented the Debug trait on some classes and added some pretty formatting utils.

### CODE REVIEW

File: chain/client-primitives/src/types.rs [hunk 1]
```diff
@@ -605,6 +605,8 @@ pub struct NetworkInfoResponse {
 }
 
 /// Status of given transaction including all the subsequent receipts.
+
```
Comment (nikurt) [inline chain/client-primitives/src/types.rs:608]:
```suggestion
```

**Review by nikurt — APPROVED** [review]
I'm not familiar with the implementation of coldDB and don't understand why endian-ness was changed. Stripping RC makes sense, but not stripping RC makes even more sense.

Thank you for the code comments. They are easy to understand.

**Comment by wacban** [comment]
> why endian-ness was changed

I believe the idea was that RockDB can handle that more efficiently. I don't know if we ever actually benchmarked it and saw any performance improvements.

File: core/store/src/db/splitdb.rs [hunk 2]
```diff
@@ -74,83 +107,99 @@ impl Database for SplitDB {
...
-    /// For cold columns returns the cold iterator.
-    /// For other columns returns the hot iterator.
+    /// The returned iterator will iterate through items in both the cold store
+    /// and the hot store. The items will be unique and sorted.
     fn iter_prefix<'a>(&'a self, col: DBCol, key_prefix: &'a [u8]) -> DBIterator<'a> {
-        match col.is_cold() {
-            false => self.hot.iter_prefix(col, key_prefix),
-            true => self.cold.iter_prefix(col, key_prefix),
+        if !col.is_cold() {
+            return self.hot.iter(col);
```
Comment (posvyatokum) [inline core/store/src/db/splitdb.rs:127]: Not iter_prefix?

**Review by posvyatokum — CHANGES_REQUESTED** [review]
Should we add iteration tests to split db also?

File: core/store/src/db/splitdb.rs (hunk 2 above)
Comment (wacban) [reply core/store/src/db/splitdb.rs:127]: Thanks, good catch!

**Review by posvyatokum — APPROVED** [review]

**Pull request merged**